TheRealGD/therealgd
# Issue 160: Moderation queue and new report system

This PR implements the moderation queue, reporting and post moderation aspects of issue #160 

This does NOT cover blocking (to the extent that the mockups state) and the Auto Moderation features.

To test this feature:

- Log in as a non moderator
- Click the report button on a submission
- Fill out the form and submit the report
- Click the block user button
- Reload the page
- You should *NOT* see the report count/approve/remove buttons (as you are not a moderator/admin)
- Follow the same process, clicking the report button on a comment.
- Again, you should not see the moderation buttons.
- Create multiple submissions and comments and report them for test cases later.
- Logout and log back in as a moderator of the forum, or an administrator.
- Navigate to the forum page and click on the Moderation Queue link in the sidebar.
- You should see two entries, one for each report that was made.
- You should also see moderation buttons for these entries.
- You can click the report count buttons to view the reports.
- You can click the accept button to clear the entry from the moderation queue without removing the submission/comment
- You can click the remove button to delete the comment (it will soft delete the comment)/submission.
- The moderation links will also be shown in the forum, outside of the moderation queue too, where they will act in the same way.
